### PR TITLE
GCP: fix config period value when the given value is below the minimum allowed period.

### LINKED ISSUE #30434: [Metricbeat] GCP module should not error when period is less than minimum
**Describe the enhancement:** Currently GCP module errors when a period less than a minimum given by GCP APIs is specified. It should not produce an error, correct the value to the minimum and produce a warning log message.

**Describe a specific use case for the enhancement or feature:** `gpc.compute` integration package specify a default period of 10s. When the integration is installed and setup to collect metrics from GCP metricbeat fails initializing data collection. This behaviour is correct but suboptimal, as debugging it requires delving into the Elastic Agent logs (identifying this issue from the log console in Kibana failed me multiple times). Given that the minimal period is known, is possible to correct this error without requiring user input.

Separately this should be addressed in the integration configuration by updating the default value **and** specifying the minimum valid value in the description.

### PR DESCRIPTION
**Pull request by Linu-Elias**
## Proposed commit message
Currently GCP module errors when a period less than a minimum given by GCP APIs is specified. It should not produce an error, correct the value to the minimum and produce a warning log message.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Disruptive User Impact

- [ ]

- Closes #30434

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -167,6 +167,8 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Normalize AWS RDS CPU Utilization values before making the metadata API call. {pull}39664[39664]
 - Fix behavior of pagetypeinfo metrics {pull}39985[39985]
 - Fix query logic for temp and non-temp tablespaces in Oracle module. {issue}38051[38051] {pull}39787[39787]
+- Fix for GCP to update the period value to default value(60s) when user specifies a value below the minimum period value in GCP.
```
Comment (gpop63) [inline CHANGELOG.next.asciidoc:170]: Remove the new line added after the changelog entry please.

```suggestion
- Set GCP metrics config period to the default (60s) when the value is below the minimum allowed period. {pull}40020[40020]
```

File: x-pack/metricbeat/module/gcp/metrics/metricset.go [hunk 2]
```diff
@@ -144,7 +144,10 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	}
 
 	if err := validatePeriodForGCP(m.Module().Config().Period); err != nil {
-		return nil, err
+		m.Logger().Warnf("Period has been updated to GCP default value as  %s", err)
```
Comment (gpop63) [inline x-pack/metricbeat/module/gcp/metrics/metricset.go:147]:
```suggestion
		m.Logger().Warnf("Period has been set to default value of 60s: %s", err)
```

Comment (Linu-Elias) [reply x-pack/metricbeat/module/gcp/metrics/metricset.go]: Done

File: CHANGELOG.next.asciidoc (hunk 1 above)
Comment (Linu-Elias) [reply CHANGELOG.next.asciidoc]: Done

**Review by efd6 — APPROVED** [review]

**Review by ishleenk17 — APPROVED** [review]

**Review by gpop63 — APPROVED** [review]

**Pull request merged**